src/test/java/org/prebid/server/settings/EnrichingApplicationSettingsTest.java

```diff
@@ -202,84 +204,74 @@ public void getAccountByIdShouldReturnFailedFutureWhenDelegateFailedAndEnforceVa
 
         // then
         assertThat(accountFuture).isFailed();
+        verifyNoInteractions(priceFloorsConfigResolver, activitiesConfigResolver);
     }
 
     @Test
-    public void getAccountByIdShouldPassOnFailureWhenDefaultAccountIsEmpty() {
+    public void getAccountByIdShouldReturnFailedFutureWhenAccountIdIsBlankAndEnforceValidAccountIsTrue() {
         // given
         target = new EnrichingApplicationSettings(
                 true,
-                0,
-                Account.builder().build(),
+                "{\"auction\": {\"banner-cache-ttl\": 100}}",
                 delegate,
                 priceFloorsConfigResolver,
-                jsonMerger);
-
-        given(delegate.getAccountById(anyString(), any())).willReturn(Future.failedFuture("Exception"));
+                activitiesConfigResolver,
+                jsonMerger,
+                jacksonMapper);
 
         // when
-        final Future<Account> accountFuture = target.getAccountById("123", timeout);
+        final Future<Account> accountFuture = target.getAccountById("", timeout);
 
         // then
         assertThat(accountFuture).isFailed();
+        verifyNoMoreInteractions(delegate);
     }
 
     @Test
-    public void getAccountByIdShouldRemoveInvalidRulesFromAccountActivitiesConfiguration() {
+    public void getAccountByIdShouldReturnEmptyAccWhenAccountIdIsBlankAndEnforceValidAccountIsFalse() {
+        // given
+        target = new EnrichingApplicationSettings(
+                false,
+                "{\"auction\": {\"banner-cache-ttl\": 100}}",
+                delegate,
+                priceFloorsConfigResolver,
+                activitiesConfigResolver,
+                jsonMerger,
+                jacksonMapper);
+
+        // when
+        final Future<Account> accountFuture = target.getAccountById("", timeout);
+
+        // then
+        assertThat(accountFuture).succeededWith(Account.builder()
+                .id("")
+                .auction(AccountAuctionConfig.builder()
+                        .bannerCacheTtl(100)
+                        .build())
+                .build());
+        verifyNoMoreInteractions(delegate);
+        verifyNoInteractions(priceFloorsConfigResolver, activitiesConfigResolver);
+    }
+
+    @Test
+    public void getAccountByIdShouldPassOnFailureWhenDefaultAccountIsEmpty() {
```

Comment:
How this test differs from `getAccountByIdShouldReturnFailedFutureWhenDelegateFailedAndEnforceValidAccountIsTrue` test?

Reply:
removed